test/Microsoft.AspNet.Mvc.TagHelpers.Test/ScriptTagHelperTest.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Logging;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    public class ScriptTagHelperTest
+    {
+        [Theory]
+        [InlineData("~/blank.js")]
+        [InlineData(null)]
+        public async Task RunsWhenRequiredAttributesArePresent(string srcValue)
+        {
+            // Arrange
+            var attributes = new Dictionary<string, object>
+                {
+                    { "asp-fallback-src", "http://www.example.com/blank.js" },
+                    { "asp-fallback-test", "isavailable()" },
+                };
+
+            if (srcValue != null)
+            {
+                attributes.Add("src", srcValue);
+            }
+
+            var context = MakeTagHelperContext(attributes);
+
+            var output = MakeTagHelperOutput("script");
+            var logger = new Logger();
+
+            // Act
+            var helper = new ScriptTagHelper()
+            {
+                Logger = logger,
+                FallbackSrc = "http://www.example.com/blank.js",
+                FallbackTestExpression = "isavailable()",
+            };
+
+            await helper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.Null(output.TagName);
+            Assert.NotNull(output.Content);
+            Assert.True(output.ContentSet);
+            Assert.Equal(0, logger.Logged.Count);
+        }
+
+        [Theory]
+        [InlineData("asp-fallback-src")]
+        [InlineData("asp-fallback-test")]
+        public async Task DoesNotRunWhenARequiredAttributeIsMissing(string attributeToRemove)
+        {
+            // Arrange
+            var attributes = new Dictionary<string, object>
+                {
+                    { "asp-fallback-src", "http://www.example.com/blank.js" },
+                    { "asp-fallback-test", "isavailable()" },
+                };
+
+            attributes.Remove(attributeToRemove);
+
+            Assert.Equal(1, attributes.Count);
+
+            var context = MakeTagHelperContext(attributes);
+
+            var output = MakeTagHelperOutput("script");
+            var logger = new Logger();
+
+            // Act
+            var helper = new ScriptTagHelper
+            {
+                Logger = logger,
```

Comment:
You're not setting either property, to avoid reflection I'd suggest creating an individual test for each attribute. Not too bad since there's only two.